metabase/metabase
# fix(sdk): show loader right after visualizing in notebook editor for the first time

Closes https://github.com/metabase/metabase/issues/47564

### Description

There is no feedback after clicking on "Visualize" for the first time when creating new questions. You can notice this in [Shoppy when creating questions](https://embedded-analytics-sdk-demo.metabase.com/admin/analytics/new/from-scratch) as the viz only shows up after the question finishes loading, which takes a long time on Shoppy.

This PR triggers `onApply` immediately when "Visualize" is clicked. It also fixes an issue where we show "Question not found" instead of a loading spinner in the QuestionVisualization view for the first time.

### How to verify

- Use either the `CreateQuestion` component in the Storybook story, or use the question creator in Shoppy with this PR.
- Click on Visualize
- It should switch to the visualize view right away. It will show a loading indicator.

### Demo

After clicking on Visualize when creating questions in Shoppy, it shows a loading indicator.

![CleanShot 2567-11-22 at 23 53 32@2x](https://github.com/user-attachments/assets/38c14a61-2772-4160-be60-32f213b5b1a2)

### Checklist
